### PR TITLE
BOM-2175 : Upgrade To Python 3.8

### PR DESCRIPTION
**Pull request by mzulqarnain1**
#### What are the relevant tickets?

- https://openedx.atlassian.net/browse/BOM-2175

#### What's this PR do?

- Dropped support for Python 2.7 , 3.5 and Django 1.11
- Upgrade To Use Python 3.8 and Django 2.2

#### How should this be manually tested?
Login to https://sgaonpy38.sandbox.edx.org and use edx-sga xblock in any course as a publisher and publish to see changes.

### CODE REVIEW

**Comment by awais786** [comment]
@mzulqarnain1 make a sandbox using the hash and verify its working. You can get help from others members for testing.

**Comment by mzulqarnain1** [comment]
@awais786 Tested these changes in https://sgaonpy38.sandbox.edx.org.

**Review by awais786 — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -8,13 +8,14 @@ passenv =
...
+    django30: Django>=3.0,<3.1
+    -r test_requirements.txt
 
 commands =
-    py.test {posargs}
+    pytest {posargs}
     coverage xml
     codecov
-    pylint edx_sga
+    pylint --py3k edx_sga
```
Comment (jmbowman) [inline tox.ini:19]: This runs just the Python 3.x compatibility checks, which I don't think we need anymore.  Probably should leave this line as it was.

Comment (mzulqarnain1) [reply tox.ini]: Removed this and fixed the quality failures that came without it.

**Comment by awais786** [comment]
@odlbot Please review this.

**Review by umarmughal824 — APPROVED** [review]
👍

**Comment by umarmughal824** [comment]
@pdpinch I have tested it after installing it on my local edX instance with the upstream, it seems to be working fine.

**Comment by pdpinch** [comment]
@mzulqarnain1 @awais786 @jmbowman do you need a PyPI release with this change?

**Comment by mzulqarnain1** [comment]
@pdpinch Not exactly, you guys can release at your own convenience.

**Pull request merged**